resize/resize3.c

```diff
@@ -133,14 +130,18 @@ int main(int argc, char *argv[])
     for (i = 0; i < biHeight; i++) {
         
         // determine the row from old to be copied
-        RGBTRIPLE* old_pixel =  (RGBTRIPLE*) (old + (int) floor(i/factor));
+        RGBTRIPLE* old_pixel =  old + (int) i/factor;
     
         //set new_pixel to the address of the beginning of current row, padding included in newRowWidth
-        RGBTRIPLE* new_pixel = (RGBTRIPLE*) (new + i*newRowWidth);
+        RGBTRIPLE* new_pixel = new + i*newRowWidth;
         
             // iterate over new image pixels in scanline (row)
             for (j = 0; j < bi.biWidth; j++) {   
-                new_pixel[j] = old_pixel[(int) floor(j/factor)];
+                // assert j/factor is no larger than og_biWidth
+                if (j/factor >= og_biWidth) {
```

Comment:
I meant actually `assert`. This is something that should literally never happen if you've implemented everything correctly. Doing it this way is going to hide errors.